sdk/pinocchio/src/sysvars/instructions.rs

```diff
@@ -36,6 +36,14 @@ where
         Instructions { data }
     }
 
+    /// Load the number of instructions in the currently executing `Transaction`.
+    #[inline(always)]
+    pub fn num_instructions(&self) -> u16 {
+        // SAFETY: The first 2 bytes of the Instructions sysvar data represents the
+        // number of instructions.
+        unsafe { *(self.data.as_ptr() as *const u16) }
```

Comment:
Since we are not enforcing alignment, might be better to use `from_le_bytes`:
```suggestion
        unsafe { u16::from_le_bytes(*(self.data.as_ptr() as *const [u8; 2])) }
```

Reply:
good call, merged!